Ginger/GingerCoreNET/RecordingLib/RecordingManager.cs

```diff
@@ -0,0 +1,198 @@
+﻿using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.Common.UIElement;
+using Amdocs.Ginger.CoreNET.Application_Models;
+using Amdocs.Ginger.Plugin.Core;
+using Amdocs.Ginger.Repository;
+using GingerCore.Actions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Amdocs.Ginger.CoreNET
+{
+    public class RecordingManager
+    {       
+        private bool CreatePOM { get; set; }
+
+        public List<POMObjectRecordingHelper> ListPOMObjectHelper { get; set; }
+        
+        public Context Context { get; set; }
+
+        public IRecord PlatformDriver { get; set; }
+
+        public ApplicationPOMModel CurrentPOM { get; set; }
+
+        public IPlatformInfo PlatformInfo { get; set; }
+
+        public RecordingManager(List<ApplicationPOMModel> lstApplicationPOM, Context context, IRecord platformDriver, IPlatformInfo pInfo)
+        {
+            try
+            {
+                PlatformInfo = pInfo;
+                PlatformDriver = platformDriver;
+                //if lstApplicationPOM == null then dont create POM or if applicationPOM.Name has some value then use the existing POM
+                //or else create new POM
+                if (lstApplicationPOM == null)
+                {
+                    PlatformDriver.LearnAdditionalDetails = false;
+                    CreatePOM = false;
+                    CurrentPOM = null;
+                }
+                else
+                {
+                    PlatformDriver.LearnAdditionalDetails = true;
+                    CreatePOM = true;
+                    CurrentPOM = lstApplicationPOM[0];
+                    ListPOMObjectHelper = new List<POMObjectRecordingHelper>();
+                    foreach (var cPom in lstApplicationPOM)
+                    {
+                        ListPOMObjectHelper.Add(new POMObjectRecordingHelper() { PageTitle = cPom.ItemName, PageURL = cPom.PageURL, ApplicationPOM = cPom });
+                    }
+                    PlatformDriver.PageChanged += PlatformDriver_PageChanged;
+                }
+
+                Context = context;
+                PlatformDriver.ElementRecorded += PlatformDriver_ElementRecorded;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Error in Recording Manager while instantiating", ex);
+            }
+        }
+
+        private POMObjectRecordingHelper GetRecordingHelperObject(string pageTitle, string pageURL, string screenShot)
```

Comment:
if i understand correctly, this "GetRecordingHelperObject()" function needs to be called "GetNewPOM()"

Reply:
renamed